python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -705,13 +721,56 @@ def get_lsst_magnitudes(self):
         """
         getter for LSST magnitudes of solar system objects
         """
-
         if not hasattr(self, 'lsstBandpassDict'):
             self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
 
         return self._magnitudeGetter(self.lsstBandpassDict, self.get_lsst_magnitudes._colnames)
 
 
+    def get_magFilter(self):
+        """
+        Generate the magnitude in the filter of the observation.
+        """
+        if not hasattr(self, 'lsstBandpassDict'):
+            self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+        mags = self._magnitudeGetter(self.lsstBandpassDict, self.lsstBandpassDict.keys())
+        mag_keys = self.lsstBandpassDict.keys()
+        magFilter = mags[mag_keys.index(self.obs_metadata.bandpass)]
+        return magFilter
+
+    def get_SNR(self):
+        """
+        Calculate the SNR for the observation, given m5 from obs_metadata and the trailing losses.
+        """
+        magFilter = self.column_by_name('magFilter')
+        bandpass = self.lsstBandpassDict[self.obs_metadata.bandpass]
+        # Get m5 for the visit
+        m5 = self.obs_metadata.m5[self.obs_metadata.bandpass]
+        # Adjust the magnitude of the source for the trailing losses.
+        dmagSNR = self.column_by_name('dmagTrailing')
+        magObj = (magFilter - dmagSNR).reshape((1, len(magFilter)))
+        if len(magObj) == 0:
+            snr = []
+        else:
+            snr, gamma = calcSNR_m5(magObj, [bandpass], [m5], self.photParams)
+            snr = snr.reshape((len(magFilter), 1))
```

Comment:
You should just be able to do 

return snr[0]

to get the numpy array corresponding to the magnitude you want